src/pilot/assign_role.py

```diff
@@ -1056,6 +1065,14 @@ def generate_osd_config_with_journals(controllers, osd_drives, ssds):
 
     return osd_config
 
+def is_nvme_drive(disk):
```

Comment:
This looks good.  Can you add a comment here that says something like:
# This method could be called with either physical or virtual disks, and only physical disks have the device_protocol attribute.  Virtual disks can never be NVMe drives, so only return True if device_protocol is present and indicates NVMe.

Reply:
Done.